benchmarks/benchmark_functions/machine_learning.py

```diff
@@ -17,69 +17,142 @@
 from numpy.random import random
 
 import pennylane as qml
-import tensorflow as tf
-import torch
 from pennylane import numpy as pnp
-from .default_settings import _core_defaults
+from .default_settings import _ml_defaults
 
 
-qml.enable_tape()
+def _machine_learning_autograd(quantum_model, data):
+	"""ML example with autograd interface."""
 
+	def hybrid_model(x, w_quantum, w_classical):
+		transformed_x = pnp.dot(w_classical, x)
+		return quantum_model(transformed_x, w_quantum)
 
-def _machine_learning_autograd(circuit, params):
-	return NotImplemented
+	def average_loss(w_quantum, w_classical):
+		c = 0
+		for x, y in data:
+			prediction = hybrid_model(x, w_quantum, w_classical)
+			c += (prediction - y) ** 2
+		return c / len(data)
 
+	n_features = len(data[0][0])
 
-def _machine_learning_tf(circuit, params):
-	return NotImplemented
+	w_quantum = pnp.array(random(size=(n_features, n_features)), requires_grad=True)
+	w_classical = pnp.array(random(size=(n_features, n_features)), requires_grad=True)
 
+	gradient_fn_wq = qml.grad(average_loss, argnum=0)
+	gradient_fn_wc = qml.grad(average_loss, argnum=1)
 
-def _machine_learning_torch(circuit, params):
-	return NotImplemented
+	for i in range(50):
+		w_quantum = w_quantum - 0.05 * gradient_fn_wq(w_quantum, w_classical)
+		w_classical = w_classical - 0.05 * gradient_fn_wc(w_quantum, w_classical)
+
+
+def _machine_learning_tf(quantum_model, data):
+	"""ML example with tensorflow interface."""
+
+	import tensorflow as tf
```

Comment:
I'm assuming if we import here, then we can run the other functions without importing tensorflow?  

Can we change `gradient.py` to import this way, so we can run autograd benchmarking without having to get tensorflow and torch installed?

Reply:
Done.